src/lib.rs

```diff
@@ -1,122 +1,100 @@
 //! dependencies.
 //!
-//! | Format | Decoding | Encoding |
-//! | ------ | -------- | -------- |
-//! | PNG    | All supported color types | Same as decoding |
-//! | JPEG   | Baseline and progressive | Baseline JPEG |
-//! | GIF    | Yes | Yes |
-//! | BMP    | Yes | RGB(8), RGBA(8), Gray(8), GrayA(8) |
-//! | ICO    | Yes | Yes |
-//! | TIFF   | Baseline(no fax support) + LZW + PackBits | RGB(8), RGBA(8), Gray(8) |
-//! | WebP   | Lossy(Luma channel only) | No |
-//! | PNM    | PBM, PGM, PPM, standard PAM | Yes |
-//! | DDS    | DXT1, DXT3, DXT5 | No |
-//! | TGA    | Yes | RGB(8), RGBA(8), BGR(8), BGRA(8), Gray(8), GrayA(8) |
-//! | farbfeld | Yes | Yes |
-//!
-//! ## Using images decoders
-//!
-//! There exists a huge variety of image formats that are concerned with efficiently encoding image
-//! pixel data and auxiliary meta data for many different purposes. The `image` library provides
-//! decoders for many common formats, depending on the active features. The best way to use them
-//! depends on your use case.
-//!
-//! * [`open`] is a very simple way to load images from the file system, automatically deducing the
-//!   format but offering little customization.
-//! * [`load_from_memory`], [`load_from_memory_with_format`] present a similar interface for images
-//!   whose encoded data is already present in memory.
-//! * [`io::Reader`] is a builder providing a superset of the functions. It offers both
-//!   customization and auto-deduction but is slightly more involved. The main benefit is that the
-//!   interface is easier to evolve.
-//! * [`ImageDecoder`] is a trait for querying meta data and reading image pixels into a generic
-//!   byte buffer. It also contains a `Read` adaptor for stream reading the pixels.
-//! * [`DynamicImage::from_decoder`] can be used for creating a buffer from a single specific or
-//!   any custom decoder implementing the [`ImageDecoder`] trait.
-//!
-//! [`open`]: fn.open.html
-//! [`load_from_memory`]: fn.load_from_memory.html
-//! [`load_from_memory_with_format`]: fn.load_from_memory_with_format.html
+//! # High level API
+//!
+//! Load images using [`io::Reader`]:
+//!
+//! ```rust,no_run
+//! # use std::io::Cursor;
+//! use image::io::Reader as ImageReader;
+//! # fn main() -> Result<(), image::ImageError> {
+//! # let bytes = vec![0u8];
+//!
+//! let img = ImageReader::open("myimage.png")?.decode()?;
+//! let img2 = ImageReader::new(Cursor::new(bytes)).decode()?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! And save them using [`save`] or [`write_to`] methods:
+//!
+//! ```rust,no_run
+//! # use std::io::Write;
+//! # use image::ImageOutputFormat;
+//! # use image::DynamicImage;
+//! # fn main() -> Result<(), image::ImageError> {
+//! # let img: DynamicImage = unimplemented!();
+//! # let img2: DynamicImage = unimplemented!();
+//! img.save("empty.jpg")?;
+//!
+//! let mut bytes: Vec<u8> = Vec::new();
+//! img2.write_to(&mut bytes, image::ImageOutputFormat::Png)?;
```

Comment:
Unfortunately not available except with the `png` feature. Actually, why is that? We don't do the same with `ImageFormat` and it's always possible to instead error. Or a method for querying this dynamically..